### PR TITLE
Replace blog link with link to GitHub user docs

### PR DESCRIPTION
**Pull request by felicitymay**
Both the GitHub user docs and the CodeQL docs currently link to an old blog post for information about security severity levels. Following research that showed many more code scanning users following the link to the blog post than CodeQL users, we're adding the information to the GitHub user docs and updating the link in the CodeQL docs.

For more information, see the linked issue and other associated pull request (which contains an internal link to a preview of the updated GitHub user docs).

### CODE REVIEW

**Review by subatoi — DISMISSED** [review]
LGTM docs-wise 👍

**Comment by felicitymay** [comment]
@github/codeql-analysis-reviewers - please check that you're happy with this minor change to the documentation.

File: docs/codeql/writing-codeql-queries/metadata-for-codeql-queries.rst [hunk 1]
```diff
@@ -49,7 +49,7 @@ The following properties are supported by all query files:
...
-| ``@security-severity``| ``<score>``               | Defines the level of severity, between 0.0 and 10.0, for queries with ``@tags security``. For more information about calculating ``@security-severity``, see the `GitHub changelog <https://gh.io/2021-07-19-codeql-security-severity>`__.                                                                                                                            |
+| ``@security-severity``| ``<score>``               | Defines the level of severity, between 0.0 and 10.0, for queries with ``@tags security``. For more information, see the `GitHub user documentation <https://docs.github.com/code-security/code-scanning/managing-code-scanning-alerts/about-code-scanning-alerts#about-alert-severity-and-security-severity-levels>`__.                                               |
```
Comment (angelapwen) [inline docs/codeql/writing-codeql-queries/metadata-for-codeql-queries.rst:52]: I don't think this links to the appropriate section: should the link be https://docs.github.com/en/code-security/code-scanning/managing-code-scanning-alerts/about-code-scanning-alerts#about-security-severity-levels?

File: docs/query-metadata-style-guide.md [hunk 2]
```diff
@@ -117,7 +117,7 @@ Alert queries (`@kind problem` or `path-problem`) support two further properties
   * `error`–an issue that is likely to cause incorrect program behavior, for example a crash or vulnerability.
   * `warning`–an issue that indicates a potential problem in the code, or makes the code fragile if another (unrelated) part of code is changed.
   * `recommendation`–an issue where the code behaves correctly, but it could be improved.
-* `@security-severity`-defines the level of severity, between 0.0 and 10.0, for queries with `@tags security`. For more information about calculating `@security-severity`, see the [GitHub changelog](https://github.blog/changelog/2021-07-19-codeql-code-scanning-new-severity-levels-for-security-alerts/).  
+* `@security-severity`-defines the level of severity, between 0.0 and 10.0, for queries with `@tags security`. For more information about how this value is calculated and then used in code scanning analysis, see [About alert severity and security severity levels](https://docs.github.com/code-security/code-scanning/managing-code-scanning-alerts/about-code-scanning-alerts#about-alert-severity-and-security-severity-levels) in the GitHub user documentation.
```
Comment (angelapwen) [inline docs/query-metadata-style-guide.md:120]: Same comment about the link 😄

**Review by angelapwen — COMMENTED** [review]
Just a quick note about the link!

File: docs/codeql/writing-codeql-queries/metadata-for-codeql-queries.rst (hunk 1 above)
Comment (felicitymay) [reply docs/codeql/writing-codeql-queries/metadata-for-codeql-queries.rst:52]: Thanks, Angela. The section title will change when I merge the related Docs PR. I'll check whether it's ready to merge yet.

Comment (angelapwen) [reply docs/codeql/writing-codeql-queries/metadata-for-codeql-queries.rst:52]: Ah, I see it! I'll approve this then as long as it's merged after the Docs site is updated 😄

**Review by angelapwen — APPROVED** [review]

File: docs/codeql/writing-codeql-queries/metadata-for-codeql-queries.rst (hunk 1 above)
Comment (felicitymay) [reply docs/codeql/writing-codeql-queries/metadata-for-codeql-queries.rst:52]: Thank you 💖

You should find that the link now works as expected 🤞

Comment (angelapwen) [reply docs/codeql/writing-codeql-queries/metadata-for-codeql-queries.rst:52]: Hm.. https://docs.github.com/en/code-security/code-scanning/managing-code-scanning-alerts/about-code-scanning-alerts#about-alert-severity-and-security-severity-levels still isn't working for me 🤔 does it need to be deployed or something?

Comment (angelapwen) [reply docs/codeql/writing-codeql-queries/metadata-for-codeql-queries.rst:52]: Oh wow, just did a refresh and it updated right after I commented 😅 working now!

**Pull request merged**